substratevm/src/com.oracle.svm.reflect/src/com/oracle/svm/reflect/serialize/hosted/SerializationFeature.java

```diff
@@ -96,13 +121,140 @@ public void duringSetup(DuringSetupAccess a) {
+        StructuredGraph graph = new StructuredGraph.Builder(debug.getOptions(), debug).method(method).build();
+        try (DebugContext.Scope ignored = debug.scope("ParsingToMaterializeLambdas")) {
+            GraphBuilderPhase lambdaParserPhase = new GraphBuilderPhase(buildLambdaParserConfig(new NoClassInitializationPlugin()));
+            HighTierContext context = new HighTierContext(GraalAccess.getOriginalProviders(), null, OptimisticOptimizations.NONE);
+            lambdaParserPhase.apply(graph, context);
+        } catch (Throwable e) {
+            throw debug.handle(e);
+        }
+        return graph;
+    }
+
+    private static Class<?> lookupForMemberField(Class<?> clazz, Object constantObject) {
+        try {
+            Field memberField = clazz.getDeclaredField("member");
+            memberField.setAccessible(true);
+            Object object = memberField.get(constantObject);
+            Method getDeclaringClass = object.getClass().getDeclaredMethod("getDeclaringClass");
+            getDeclaringClass.setAccessible(true);
+            return (Class<?>) getDeclaringClass.invoke(object);
+        } catch (NoSuchFieldException | IllegalAccessException | NoSuchMethodException | InvocationTargetException exception) {
+            return null;
+        }
+    }
+
+    private static Class<?> lookupForLambdaClass(Class<?> clazz, Object constantObject, String fieldName) {
+        try {
+            Field instanceClassField = clazz.getDeclaredField(fieldName);
+            instanceClassField.setAccessible(true);
+            return (Class<?>) instanceClassField.get(constantObject);
+        } catch (IllegalAccessException | NoSuchFieldException ignored) {
+            return null;
+        }
+    }
+
+    private static Class<?> getLambdaClassFromConstantNode(ConstantNode constantNode) {
+        Constant constant = constantNode.getValue();
+        Object constantObject;
+
+        try {
+            Field objectField = constant.getClass().getDeclaredField("object");
+            objectField.setAccessible(true);
+            constantObject = objectField.get(constant);
+        } catch (IllegalAccessException | NoSuchFieldException ignored) {
+            return null;
+        }
+
+        Class<?> clazz = constantObject.getClass();
+        Class<?> lambdaClass = lookupForMemberField(clazz, constantObject);
+
+        if (lambdaClass == null) {
+            if (JavaVersionUtil.JAVA_SPEC >= 17) {
+                lambdaClass = lookupForLambdaClass(clazz, constantObject, "instanceField");
+            } else {
+                lambdaClass = lookupForLambdaClass(clazz, constantObject, "staticBase");
+            }
+        }
+
+        if (lambdaClass == null) {
+            return null;
+        }
+
+        return lambdaClass.getName().contains(LambdaUtils.LAMBDA_CLASS_NAME_SUBSTRING) ? lambdaClass : null;
+    }
+
+    private static void traverseGraph(StructuredGraph graph) {
```

Comment:
`traverseConstants` would fit better. Also, it doesn't say what you do with them.

Reply:
Renamed as suggested.